pkgs/by-name/ll/llama-swap/package.nix

```
@@ -0,0 +1,104 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  npmHooks,
+  fetchNpmDeps,
+  nodejs,
+}:
+
+let
+  canExecute = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+in
+buildGoModule (finalAttrs: {
+  pname = "llama-swap";
+  version = "149";
+
+  src = fetchFromGitHub {
+    owner = "mostlygeek";
+    repo = "llama-swap";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-QTv12wTMEYeoJt4sO9L/oueXzDH7JyEO0hT4GXYCJBU=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      # '0000-00-00T00:00:00Z'
+      date -u -d "@$(git log -1 --pretty=%ct)" "+'%Y-%m-%dT%H:%M:%SZ'" > $out/SOURCE_DATE_EPOCH
+      find "$out" -name .git -print0 | xargs -0 rm -rf
```

```suggestion
      rm -rf .git
```
the PWD is already the root dir

since the result will be the same, hash shouldn't change

---

this is a common pattern.

```
λ rg "\-name \.git \-print0" --count-matches | awk -F: '{s+=$2} END {print s}'
45
```

https://github.com/NixOS/nixpkgs/commit/9d7932e349942db74ec2c3a7030085c77d7a3716#diff-2781420d92792cbc2471e058d776f10a032c5519459f07df791f395b9bde29f5R12-R22

It'll find any `.git` dirs across the whole output and is the exact command ran if you don't enable `leaveDotGit`.
Since the intention is to: grab extra info, then go back to `leaveDotGit = false;` behavior, I'd rather keep it

https://github.com/NixOS/nixpkgs/blob/67bb48c4c8e327417d6d5aa7e538244b209e852b/pkgs/build-support/fetchgit/nix-prefetch-git#L348

---

fair enough, in case there are submodules. consider this convo resolved